src/components/capture/create/index.tsx

```diff
@@ -135,130 +134,108 @@ function CaptureCreation() {
         selectors.form.exitWhenLogsClose
     );
 
-    // Local state
-    const [catalogResponse, setCatalogResponse] = useState<any | null>(null);
-
-    const cleanUpEditor = () => {
-        if (Object.keys(resourcesFromEditor).length > 0) {
-            clearResourcesFromEditor();
-        }
-    };
+    //Editor state
+    const draftId = useEditorStore(editorStoreSelectors.draftId);
+    const setDraftId = useEditorStore(editorStoreSelectors.setDraftId);
 
-    const exit = () => {
-        cleanUpEditor();
-
-        resetState();
-
-        navigate('/captures');
-    };
-
-    const discovers = {
-        done: (discoversSubscription: RealtimeSubscription) => {
+    const helpers = {
+        callFailed: (formState: any, subscription?: RealtimeSubscription) => {
+            const setFailureState = () => {
+                setFormState({
+                    status: CaptureCreationFormStatus.IDLE,
+                    exitWhenLogsClose: false,
+                    saveStatus: intl.formatMessage({
+                        id: 'captureCreation.status.failed',
+                    }),
+                    ...formState,
+                });
+            };
+            if (subscription) {
+                helpers
+                    .doneSubscribing(subscription)
+                    .then(() => {
+                        setFailureState();
+                    })
+                    .catch(() => {});
+            } else {
+                setFailureState();
+            }
+        },
+        doneSubscribing: (subscription: RealtimeSubscription) => {
             return supabaseClient
-                .removeSubscription(discoversSubscription)
+                .removeSubscription(subscription)
                 .then(() => {
                     setFormState({
                         status: CaptureCreationFormStatus.IDLE,
                     });
                 })
                 .catch(() => {});
         },
-        waitForFinish: () => {
-            cleanUpEditor();
+        exit: () => {
+            resetState();
+
+            navigate('/captures');
+        },
+        jobFailed: (errorTitle: string) => {
+            setFormState({
+                error: {
+                    title: errorTitle,
+                },
+                saveStatus: intl.formatMessage({
+                    id: 'captureCreation.status.failed',
+                }),
+            });
+        },
+    };
+
+    const waitFor = {
+        base: (query: any, success: Function, failureTitle: string) => {
```

Comment:
Created a base waitFor that the other two share. We'll expand on this eventually.